manifests/resource/location.pp

```diff
@@ -393,41 +388,64 @@
     warning('The $fastcgi_script parameter is deprecated; please use $fastcgi_param instead to define custom fastcgi_params!')
   }
 
-  $vhost_sanitized = regsubst($vhost, ' ', '_', 'G')
+  $server_sanitized = regsubst($server, ' ', '_', 'G')
   if $::nginx::confd_only {
-    $vhost_dir = "${::nginx::conf_dir}/conf.d"
+    $server_dir = "${::nginx::conf_dir}/conf.d"
   } else {
-    $vhost_dir = "${::nginx::conf_dir}/sites-available"
+    $server_dir = "${::nginx::conf_dir}/sites-available"
   }
 
-  $config_file = "${vhost_dir}/${vhost_sanitized}.conf"
+  $config_file = "${server_dir}/${server_sanitized}.conf"
 
   $location_sanitized_tmp = regsubst($location, '\/', '_', 'G')
   $location_sanitized = regsubst($location_sanitized_tmp, '\\\\', '_', 'G')
 
+  # Use proxy or fastcgi template if $proxy is defined, otherwise use directory template.
+  if ($proxy != undef) {
+    $content_real = template('nginx/server/locations/proxy.erb')
+  } elsif ($location_alias != undef) {
+    $content_real = template('nginx/server/locations/alias.erb')
+  } elsif ($stub_status != undef) {
+    $content_real = template('nginx/server/locations/stub_status.erb')
+  } elsif ($fastcgi != undef) {
+    $content_real = template('nginx/server/locations/fastcgi.erb')
+  } elsif ($uwsgi != undef) {
+    $content_real = template('nginx/server/locations/uwsgi.erb')
+  } elsif ($www_root != undef) {
+    $content_real = template('nginx/server/locations/directory.erb')
+  } elsif ($try_files != undef) {
+    $content_real = template('nginx/server/locations/try_files.erb')
+  } else {
+    $content_real = template('nginx/server/locations/empty.erb')
+  }
+
```

Comment:
This block shouldn't have been added back. I think you'll have to do the search/replace on the latest code from master.